### ISSUE TITLE
HL-1202; PDF Printing

### ISSUE THREAD

**Issue by lordzim** [pdewacht/brlaser#49]
Hi, i’m really new on Linux. My printer is a Brother HL-1202. Currently using a raspberry pi zero W to connect my Brother HL but having trouble to use it with CUPS.

I clone the git and did everything right (I guess), Cups shows Brlaser V6, and currently using HL-1200 series.

I manage to print the Test Print, and some text documents, and vector files. But i’m having a hard time to print ANY PDF file.

This error shows up 
Fri Oct 11 21:38:29 2019 
"/var/spool/cups/d00006-001 (object 20 0): object has offset 0"

More informations about my printer, if this is helpful.

uri = usb://Brother/HL-1200%20series?serial=F9N305579
        class = direct
        info = Brother HL-1200 series
        make-and-model = Brother HL-1200 series
        device-id = MFG:Brother;CMD:PJL,HBP;MDL:HL-1200 series;CLS:PRINTER;CID:Brother Laser Type3;
        location = 

If you need some log, please, write it down step-by-step how to capture it, i’m really new on this.

Thanks for the help.

Brlaser project saved me, i wasn’t able to print anything without using this driver before.

**Comment by pdewacht** [comment]
I don't think this is a driver issue. If only some kinds of files fail, it must be some earlier step in the print process.

I'm guessing that error message comes from the CUPS error log? If you give me the full log file, I can check if I spot anything. Also, which distribution are you using?

**Comment by lordzim** [comment]
Raspbian Buster with desktop and recommended software
Version:September 2019
Release date:2019-09-26
Kernel version:4.19
Size: 2541 MB

From raspberry.org

This message comes from JOBS at Cups web interface

How do i send the full log to you, it’s safe to drop down bellow as attach over here? I manage to grab by cupsctl --debug-logging

**Comment by pdewacht** [comment]
You can attach the log here, or mail them to me at pdewacht@gmail.com. They might have some small bits of personal info, like the names of the files you tried to print.

**Comment by neilpa** [comment]
I've recently had a couple of PDFs either partially or fully fail to print with the the same `object has offset 0` error but have printed plenty of others. [This is one example](https://tinaja.com/glib/nonlingr.pdf) where the first 3 pages prints before it stops.

I'm on MX Linux and here's my printer details.

```
$ sudo lpinfo --include-schemes usb -l -v
Device: uri = usb://Brother/HL-L2300D%20series?serial=**********
        class = direct
        info = Brother HL-L2300D series
        make-and-model = Brother HL-L2300D series
        device-id = MFG:Brother;CMD:PJL,HBP;MDL:HL-L2300D series;CLS:PRINTER;CID:Brother Laser Type1;
        location = 
```

Let me know if you're still interested in a full debug log and I'll shoot you an email.

**Comment by lordzim** [comment]
Hi
Glade you still investigating this issue, but I'm not a Linux expert or
code developer I'm afraid I can't contribute enough with this error, the
very last I could do it's hope for the best.

I still having some issue printing some PDF. Most of them involving
CamScanner PDF generated files by my android phone.

If you need something that I can test, just let me know with some
instructions. I would be happy to help.

I own a Raspberry Pi Zero W and Raspberry Pi 4 for some tests and personal
projects.



Em 21 de fev de 2020 à(s) 16:18, Neil Pankey <notifications@github.com>
escreveu:

I've recently had a couple of PDFs either partially or fully fail to print
with the the same object has offset 0 error but have printed plenty of
others. This is one example <https://tinaja.com/glib/nonlingr.pdf> where
the first 3 pages prints before it stops.

I'm on MX Linux and here's my printer details.

$ sudo lpinfo --include-schemes usb -l -v
Device: uri = usb://Brother/HL-L2300D%20series?serial=**********
        class = direct
        info = Brother HL-L2300D series
        make-and-model = Brother HL-L2300D series
        device-id = MFG:Brother;CMD:PJL,HBP;MDL:HL-L2300D
series;CLS:PRINTER;CID:Brother Laser Type1;
        location =

Let me know if you're still interested in a full debug log and I'll shoot
you an email.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/pdewacht/brlaser/issues/49?email_source=notifications&email_token=AFH4ZH2RGM43QXFC32VOHBDREASG7A5CNFSM4JAARTZ2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMTZE7Y#issuecomment-589795967>,
or unsubscribe

**Comment by pdewacht** [comment]
Hi @lordzim and @neilpa 

I tried printing neilpa's file, but I couldn't reproduce your problem.

I did figure out that the error message ("object has offset 0") originates in qpdf, which is used by cups-filters for some PDF processing steps. This means that the root cause might exist in qpdf, in cups-filters, or perhaps in the program you're using to print the PDF.

Without being able to reproduce the problem, there's not much I can do though. Can you tell me which program you use to print the PDFs? Additional examples of troublesome files might be useful, as well as CUPS debug logs. The output of `dpkg -l` as well (both of you look like you use Debian-derived distributions, this command will list the version of every package on your system).

**Comment by md-121** [comment]
Hi everyone,

I had the same issue with the `printer-driver-brlaser` package, which seems to install brlaser v4.
I tried to print the PDF @neilpa linked and some of my own ones.
When I selected to scale down the PDF to 50% before printing, it worked fine.
After installing brlaser v6 from GitHub, printing works without the before mentioned 50% downscaling.

My printer is a Brother DCP-1512.
I use a Raspberry Pi 1 Mod. B+ with Raspberry Pi OS Buster 32-bit.
The printer is set up in CUPS and I print using macOS Preview.
Using Adobe Reader under macOS still doesn't work for me.

**Comment by danielristic** [comment]
I'm using a Brother HL-1110 and I too have the problem where some PDFs won't print with the "object has offset 0" error.

I wanted to try updating to brlaser v6 but when I try `sudo apt-get upgrade printer-driver-brlaser` it tells me that I'm "already on the newest version (4-1)" and downloading the github repository and doing `cmake .` doesn't work either (I believe I installed the Pi OS Lite version that does not include the necessary stuff for compiling).

What would be the easiest way to try the updated version to see if it fixes my issue ?

**Comment by mtoigo** [comment]
I had this issue and the following solved it:

sudo apt update && sudo apt full-upgrade
apt-get remove printer-driver-brlaser
apt-get upgrade qpdf cups-filters
#Follow the instructions in the readme to build from source https://github.com/pdewacht/brlaser

**Comment by lineker** [comment]
@mtoigo I tried your steps but it didn't work for me. I still get `/var/spool/cups/d00273-001 (object 22 0): object has offset 0` with some files. I can't print [this file](https://absg.adventist.org/assets/public/files/lessons/2022/2Q/TE/PDFs/ETQ222_11.pdf). Any other ideas? I have a HL-L2320D

**Comment by jameswillhoite** [comment]
@mtoigo This worked for my HL-L2300D. I followed your instructions and now I'm printing. My RPi4 did not have pdf installed. I had to install that, and the cups-dev package as required via the Readme. All works great now.
